sentry-android-core/src/main/java/io/sentry/android/core/NetworkBreadcrumbsIntegration.java

```diff
@@ -0,0 +1,230 @@
+        logger.log(SentryLevel.DEBUG, "NetworkBreadcrumbsIntegration not installed.");
+        return;
+      }
+      logger.log(SentryLevel.DEBUG, "NetworkBreadcrumbsIntegration installed.");
+      addIntegrationToSdkVersion();
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (networkCallback != null) {
+      ConnectivityChecker.unregisterNetworkCallback(
+          context, logger, buildInfoProvider, networkCallback);
+      logger.log(SentryLevel.DEBUG, "NetworkBreadcrumbsIntegration remove.");
+    }
+    networkCallback = null;
+  }
+
+  @SuppressLint("ObsoleteSdkInt")
+  @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+  static final class NetworkBreadcrumbsNetworkCallback extends ConnectivityManager.NetworkCallback {
+    final @NotNull IHub hub;
+    final @NotNull BuildInfoProvider buildInfoProvider;
+
+    @Nullable Network currentNetwork = null;
+
+    @Nullable NetworkCapabilities lastCapabilities = null;
+
+    NetworkBreadcrumbsNetworkCallback(
+        final @NotNull IHub hub, final @NotNull BuildInfoProvider buildInfoProvider) {
+      this.hub = Objects.requireNonNull(hub, "Hub is required");
+      this.buildInfoProvider =
+          Objects.requireNonNull(buildInfoProvider, "BuildInfoProvider is required");
+    }
+
+    @Override
+    public void onAvailable(final @NonNull Network network) {
+      if (network.equals(currentNetwork)) {
+        return;
+      }
+      final Breadcrumb breadcrumb = createBreadcrumb("networkAvailable");
+      hub.addBreadcrumb(breadcrumb);
+      currentNetwork = network;
+      lastCapabilities = null;
+    }
+
+    @Override
+    public void onCapabilitiesChanged(
+        final @NonNull Network network, final @NonNull NetworkCapabilities networkCapabilities) {
+      if (!network.equals(currentNetwork)) {
+        return;
+      }
+      final @Nullable NetworkBreadcrumbConnectionDetail connectionDetail =
+          getNewConnectionDetails(lastCapabilities, networkCapabilities);
+      if (connectionDetail == null) {
+        return;
+      }
+      lastCapabilities = networkCapabilities;
+      final Breadcrumb breadcrumb = createBreadcrumb("networkCapabilitiesChanged");
+      breadcrumb.setData("download_bandwidth", connectionDetail.downBandwidth);
+      breadcrumb.setData("upload_bandwidth", connectionDetail.upBandwidth);
+      breadcrumb.setData("vpn_active", connectionDetail.isVpn);
+      breadcrumb.setData("network_type", connectionDetail.type);
+      if (connectionDetail.signalStrength != 0) {
+        breadcrumb.setData("signal_strength", connectionDetail.signalStrength);
+      }
+      Hint hint = new Hint();
+      hint.set("data", connectionDetail);
+      hub.addBreadcrumb(breadcrumb, hint);
+    }
+
+    @Override
+    public void onLost(final @NonNull Network network) {
+      if (!network.equals(currentNetwork)) {
+        return;
+      }
+      final Breadcrumb breadcrumb = createBreadcrumb("networkLost");
```

Comment:
`l`: what do you think of changing to uppercase-snakecase (NETWORK_LOST/AVAILABLE)? This way we would align with other system breadcrumbs